skyportal/models.py

```diff
@@ -1951,6 +1956,30 @@ class Taxonomy(Base):
     )
 
 
+def taxonomy_update_delete_logic(cls, user_or_token):
+
+    if len({'Delete taxonomy', 'System admin'} & set(user_or_token.permissions)) == 0:
+        # nothing accessible
+        return restricted.query_accessible_rows(cls, user_or_token)
```

Comment:
This function generates the query for taxonomies  that the current user can delete. L1963 says, if the querying user doesn't have System admin or Delete taxonomy acl, then no taxonomies are delete accessible. Otherwise, it says that the only taxonomies that can be deleted are those that have no associated classifications, preventing situations where classifications get deleted in a cascade when their parent taxonomy is deleted.

Reply:
Is this written up adequately in the docs and comments?  If not, we need to do that now, since Danny is right here.